### PR TITLE
Fixes overlay glitch in clipboards and allows mice to be butchered

### PR DESCRIPTION
**Pull request by ktstudios2000**
As the title suggests, papers no longer disappear once added to a clipboard, and dead mice, the reagent produced as a result of killing them, can now be butchered. The code for the latter did exist, but the functionality did not, as only living mobs can use the butchering proc.

:cl: Youbar
fix: clipboards now display held papers properly
fix: mice can be butchered
/:cl:

### CODE REVIEW

File: code/modules/paperwork/clipboard.dm [hunk 1]
```diff
@@ -27,12 +27,14 @@
...
+	var/list/dat = list()
 	if(toppaper)
-		add_overlay(toppaper.icon_state)
-		copy_overlays(toppaper)
+		dat += toppaper.icon_state
+		dat += toppaper.overlays.Copy()
 	if(haspen)
-		add_overlay("clipboard_pen")
-	add_overlay("clipboard_over")
+		dat += ("clipboard_pen")
```
Comment (ninjanomnom) [inline code/modules/paperwork/clipboard.dm]: You left in some parentheses

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 2]
```diff
@@ -113,5 +113,16 @@
...
+/obj/item/reagent_containers/food/snacks/deadmouse/attackby(obj/item/I, mob/user, params)
+	if(I.is_sharp() && user.a_intent == INTENT_HARM)
+		if(isturf(loc))
+			new /obj/item/reagent_containers/food/snacks/meat/slab/mouse(loc)
+			to_chat(user, "<span class='notice'>You butcher [src].</span>")
+			qdel(src)
+		else
+			to_chat(user, "<span class='warning'>You need to put [src] on a surface to butcher it!</span>")
+	else
+..()
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Indentation

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 3]
```diff
@@ -113,5 +113,16 @@
 	foodtype = GROSS | MEAT | RAW
 	grind_results = list("blood" = 20, "liquidgibs" = 5)
 
+/obj/item/reagent_containers/food/snacks/deadmouse/attackby(obj/item/I, mob/user, params)
+	if(I.is_sharp() && user.a_intent == INTENT_HARM)
+		if(isturf(loc))
+			new /obj/item/reagent_containers/food/snacks/meat/slab/mouse(loc)
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: You should be making use of the butcher component

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
These really should be separate prs

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 3 above)
Comment (Youbar) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm]: The butcher proc only exists for living mobs. This workaround allows for the duality of having a mouse as a food item, and being able to slice the mouse into a fillet.

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 2 above)
Comment (Youbar) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm]: I believe I've fixed the indentation, but I'm not entirely sure. The misplace ..() was about all I could identify as being the issue.

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 4]
```diff
@@ -113,5 +113,16 @@
...
+			to_chat(user, "<span class='notice'>You butcher [src].</span>")
+			qdel(src)
+		else
+			to_chat(user, "<span class='warning'>You need to put [src] on a surface to butcher it!</span>")
+	else
+		..()
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]:
```suggestion
		return ..()
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 3 above)
Comment (ninjanomnom) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm]: Then make it work for this as well if you need to

**Comment by optimumtact** [comment]
In future please make two pr's if you have two unrelated items.

we dont' charge you per PR

**Pull request merged**